include/swift/Reflection/ReflectionContext.h

```diff
@@ -202,7 +200,7 @@ class ReflectionContext
                                                RangeEnd - RangeStart);
 
     auto findMachOSectionByName = [&](std::string Name)
-        -> std::pair<std::pair<const char *, const char *>, uint32_t> {
+        -> std::pair<std::pair<const char *, const char *>, uintptr_t> {
```

Comment:
I seems to me like it'd be more robust to use `uint64_t` here, to ensure the code works with any host/target architecture combination. An assertion isn't going to protect anything in a release build of these libraries.

Reply:
Agree. I was trying to not change `ReflectionInfo` because of this, but if you think it would be better in the long term, I will try to figure out how far this change has to be propagated.

Reply:
It seems to me like a good idea for ReflectionInfo to use uint64_t instead of uintptr_t as well, yeah.